pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseTaskExecutor.java

```diff
@@ -56,4 +57,12 @@ protected Schema getSchema(String tableName) {
     Preconditions.checkState(schema != null, "Failed to find schema for table: %s", tableName);
     return schema;
   }
+
+  protected long getSegmentCrc(String tableNameWithType, String segmentName) {
+    OfflineSegmentZKMetadata segmentZKMetadata = ZKMetadataProvider
+        .getOfflineSegmentZKMetadata(MINION_CONTEXT.getHelixPropertyStore(), tableNameWithType, segmentName);
```

Comment:
It doesn't have to be an offline segment. Might be good to check the table type first?

Reply:
done